GitUI/UserControls/RevisionGrid/FilterInfo.cs

```diff
@@ -440,10 +447,16 @@ private void GetBranchRevisionFilter(ArgumentBuilder filter, Lazy<string> curren
                 AddFirstStashRef();
 
                 // If BranchFilter contains wildcards, "--branches=" must be prepended.
-                // Use the simple branch filter if without wildcards (must be Git reference)
+                // Use the simple branch filter if without wildcards (must be Git revision)
                 // in order to avoid git adding implicit /* to the "--branches=" filter.
-                bool useSimpleBranchFilter = BranchFilter.IndexOfAny(new[] { '?', '*', '[' }) == -1;
-                filter.Add(useSimpleBranchFilter ? BranchFilter : $"--branches={BranchFilter}");
+                // Also add apparent options
+                foreach (string branch in BranchFilter.Split((char[])null, StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries))
```

Comment:
I have noticed the draft state after I wrote the comment:
Is this a one-liner for `.Trim` + `!.IsNullorEmpty`?
If really want to split, then care needs to be taken of arguments containing strings in `"` or `'`...

Reply:
Draft due to dependencies, not contents.

Split is on whitespace, null is the default. 

Space cannot occur in git revisions but quotes could be cleaned.

Reply:
You can no undraft it :)